lib/unique_within_region_validator.rb

```diff
@@ -14,6 +14,8 @@
 class UniqueWithinRegionValidator < ActiveModel::EachValidator
   def validate_each(record, attribute, value)
     return if value.nil?
+    return unless record.new_record? || record.changed.include?(attribute.to_s)
```

Comment:
why the `new_record?`.  That's when we need to verify uniqueness the most.

Reply:
> when a record is created, or when the name has changed.

If a record is new, I think changed is still marked dirty for that column, so I think you only have to check the dirty values.

Reply:
run this method if it is a new record
run this method if it is an old record and this field has changed
skip this unless it is new or it is changed

may read better if:

skip this if it is not new and it has not changed

```ruby
    return if !record.new_record? && !record.changed.include?(attribute.to_s)
```

---

Are you thinking run this if it is changed?
the new record case will be changed
or if the new record case is blank, the presence filter will catch that case for us. Besides, if it is blank, we probably don't want to run the validation for uniqueness

so

```suggestion
    return unless record.changed.include?(attribute.to_s)
```

Reply:
Yes, I _think_ that should still work on new records, but you should try it out (or have specs that prove it).  Alternatively, something like `record.send("#{attribute}_changed?")`  I'm not sure if the `*_changed?` methods do the same things as `.changed.include?`

Reply:
Very good find. Perusing the code says they are quite different.
even `attribute_changed?` has some false positives when a value changes away and back to the original value.
I would prefer not to construct a method name here and just use a method that takes an attribute, but the _changed version just plain works better and handles more cases